python/ray/data/dataset.py

```diff
@@ -101,20 +101,29 @@
 
 @PublicAPI
 class Dataset(Generic[T]):
-    """Implements a distributed Arrow dataset.
+    """A Dataset is a distributed data collection for data loading and processing.
 
-    Datasets are implemented as a list of ``ObjectRef[Block]``. The block
-    also determines the unit of parallelism. The default block type is the
-    ``pyarrow.Table``. Arrow-incompatible objects are held in ``list`` blocks.
+    Datasets are implemented as a list of `ObjectRef[Block]`, where each block
+    holds an ordered collection of items, representing a shard of the overall
+    data collection. The block can be either a `pyarrow.Table`, or Python list.
+    The block also determines the unit of parallelism.
+
+    Datasets can be created in multiple ways: from synthetic data via `range_.*()`
+    APIs, from existing memory data via `from_*()` APIs, or from external storage
+    systems such as local disk, S3, HDFS etc. via the `read_*()` APIs. The
+    (potentially processed) Dataset can be saved back to external storage systems via
+    the `write_*()` APIs.
+
+    Datasets supports parallel processing at scale: transformations such as
```

Comment:
This is great! Can we maybe add 1-2 examples of how to create one here as well?

Reply:
Good point, I like code examples in comment. Done.